src/client.js

```diff
@@ -37,59 +40,42 @@ function podmanMonitor(name, method, args, callback, system) {
 }
 
 export function streamEvents(system, callback) {
-    return new Promise((resolve, reject) => {
-        podmanMonitor("libpod/events", "GET", {}, callback, system)
-                .then(reply => resolve(JSON.parse(reply)))
-                .catch(reject);
-    });
+    return podmanMonitor("libpod/events", "GET", {}, callback, system)
+            .then(reply => JSON.parse(reply));
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1339-20230706-052110-5d572290-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)